### PR TITLE
Fix bug and write test for job remapping

### PR DESCRIPTION
**Pull request by simonbray**
Inevitably, https://github.com/galaxyproject/bioblend/pull/343 does not work, because I didn't write a test for the job remapping.

### CODE REVIEW

File: bioblend/_tests/TestGalaxyJobs.py [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 from . import (
     GalaxyTestBase,
     test_util
 )
 
 
 class TestGalaxyJobs(GalaxyTestBase.GalaxyTestBase):
+    def setUp(self):
+        super().setUp()
+        self.history_id = self.gi.histories.create_history(name='TestJobRerun')['id']
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyJobs.py:13]:
```suggestion
        self.history_id = self.gi.histories.create_history(name='TestGalaxyJobs')['id']
```

File: bioblend/_tests/TestGalaxyJobs.py [hunk 2]
```diff
@@ -23,16 +34,67 @@ def test_run_and_rerun_random_lines(self):
...
 
         original_output = self.gi.tools.run_tool(
-            history_id=history_id,
+            history_id=self.history_id,
             tool_id="random_lines1",
             tool_inputs=tool_inputs,
             input_format='21.01'
         )
         original_job_id = original_output['jobs'][0]['id']
+        self._wait_job(original_job_id)
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyJobs.py:43]: This can be postponed a bit, I think.
```suggestion
```

File: bioblend/_tests/TestGalaxyJobs.py [hunk 3]
```diff
@@ -23,16 +34,67 @@ def test_run_and_rerun_random_lines(self):
...
         build_for_rerun = self.gi.jobs._build_for_rerun(original_job_id)
         self.assertEqual(build_for_rerun['state_inputs']['seed_source']['seed'], 'asdf')
 
         rerun_output = self.gi.jobs.rerun_job(original_job_id)
         rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
         original_output_content = self.gi.datasets.download_dataset(original_output['outputs'][0]['id'])
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyJobs.py:49]:
```suggestion
        rerun_job_id = rerun_output['jobs'][0]['id']
        self._wait_job(original_job_id)
        original_output_content = self.gi.datasets.download_dataset(original_output['outputs'][0]['id'])
        self._wait_job(rerun_job_id)
        rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
```

File: bioblend/_tests/TestGalaxyJobs.py (hunk 2 above)
Comment (simonbray) [reply bioblend/_tests/TestGalaxyJobs.py:43]: I think that `_build_for_rerun` on the next line won't work properly without this. I added the rest of your changes.

**Comment by simonbray** [comment]
Okay, the test is simplified a bit, but the important check is still there: https://github.com/galaxyproject/bioblend/blob/a2b75cb88004ff9460cefd0cc4c8396f434adc2c/bioblend/_tests/TestGalaxyJobs.py#L92

It's a bit concerning that the old tests were randomly failing. I could not reproduce this locally.

File: bioblend/_tests/TestGalaxyJobs.py (hunk 2 above)
Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyJobs.py:43]: I don't think the job needs to be completed to be re-runnable?

File: bioblend/_tests/TestGalaxyJobs.py [hunk 4]
```diff
@@ -33,6 +44,57 @@ def test_run_and_rerun_random_lines(self):
...
+            self._wait_job(job_id)
+
+        # check if workflow output has completed
+        workflow_output = [dataset for dataset in self.gi.histories.show_history(new_history_id, contents=True) if dataset['name'] == 'paste_output'][0]
+        self.assertEqual(workflow_output['state'], 'ok')
+        self.assertEqual(workflow_output['hid'], 2)
+        self.assertEqual(workflow_output['id'], history_contents[1]['id'])
+        self._wait_and_verify_dataset(workflow_output['id'], b'line 1\tline 1\n')
+
+    def _wait_job(self, job_id):
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyJobs.py:95]: Move to `bioblend/galaxy/jobs/__init__.py` and expose it?

File: bioblend/_tests/TestGalaxyJobs.py [hunk 5]
```diff
@@ -33,6 +44,57 @@ def test_run_and_rerun_random_lines(self):
...
+        }
+        inv = self.gi.workflows.invoke_workflow(wf['id'], wf_inputs)
+        for _ in range(120):
+            inv = self.gi.invocations.show_invocation(inv['id'])
+            if inv['steps']:
+                for step in inv['steps']:
+                    if step['job_id']:
+                        self._wait_job(step['job_id'])
+                break
+            time.sleep(.5)
```
Comment (nsoranzo) [inline bioblend/_tests/TestGalaxyJobs.py:71]: Probably it's more efficient to wait for the invocation to be scheduled and then wait for the jobs, so maybe wait for the merge of #369 .

File: bioblend/_tests/TestGalaxyJobs.py (hunk 4 above)
Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyJobs.py:95]: I've added a longer version of this method to `JobsClient` in [`367bbef` (#369)](https://github.com/galaxyproject/bioblend/pull/369/commits/367bbefac0d7a8eb2621cc160bed166275720a75)

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**